src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -0,0 +1,371 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This type is responsible for adding debugging sequence points for the executable code.
+    /// It can be combined with other <see cref="Instrumenter"/>s. Usually, this class should be 
+    /// the root of the chain in order to ensure sound debugging experience for the instrumented code.
```

Comment:
I assume this means that sequence points transformation is typically applied after all other changes. I am not sure comment is clear enough about this part.